libs/oef-core/src/comms/public_key_utils.cpp

```diff
@@ -18,16 +18,14 @@
 
 #include "oef-core/comms/public_key_utils.hpp"
 
-// static constexpr char const *LOGGING_NAME_PK = "PublicKeyUtils";
-
 RSAKey::RSAKey(const EvpPublicKey &evp_pk)
   : rsa_{nullptr}
 {
   ERR_load_crypto_strings();
   rsa_ = EVP_PKEY_get1_RSA(evp_pk.native_handle());
   if (rsa_ != nullptr)
   {
-    std::runtime_error(ERR_error_string(ERR_get_error(), NULL));
+    std::runtime_error(ERR_error_string(ERR_get_error(), nullptr));
```

Comment:
Is it to be thrown? `runtime_error` is a class.

Reply:
Huh, good spot. @qati @engineerkatie are we missing a `throw` here?

Reply:
yes, that is correct
it should `throw` if `rsa_ == NULL`
btw, the file is months-old outdated.
updated files are on this branch https://github.com/fetchai/research-consensus/pull/5

Reply:
Fixed @lrahmani